File: DAY02/P완주하지못한선수/P완주하지못한선수_박예린.py
def solution(participant, completion):
    dict = {}
    sum = 0

    # 참가자 hash dictionary 초기화
    for p in participant:
        dict[hash(p)] = p # key: hash(p), value: p
        sum += hash(p)

    for c in completion:
        if(dict[hash(c)] in completion):
            sum -= hash(c)

    # 단 한 명의 선수를 제외하고는 모든 선수가 마라톤을 완주
    answer = dict[sum]
    
    return answer

File: DAY02/P완주하지못한선수/P완주하지못한선수_박예린.java
package P완주하지못한선수;

import java.util.HashMap;
import java.util.Map;

class P완주하지못한선수_박예린 {
    public static String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> map = new HashMap<>();

        for(String player : participant) {
            map.put(player, map.getOrDefault(player, 0) + 1);
        }

        for(String player : completion) {
            map.put(player, map.get(player) - 1);
        }


        for(Map.Entry<String, Integer> entry : map.entrySet()) {
            if(!entry.getValue().equals(0)) {
                answer = entry.getKey();
                break;
            }
        }

        return answer;
    }

    public static void main(String[] args) {
        String[] participant = {"mislav", "stanko", "mislav", "ana"};
        String[] completion = {"stanko", "ana", "mislav"};
        System.out.println(solution(participant, completion));

    }
}